pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -25,68 +25,53 @@ import (
 	"time"
 
 	"google.golang.org/grpc"
+	grpccodes "google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
-	"k8s.io/apimachinery/pkg/types"
+	grpcstatus "google.golang.org/grpc/status"
 	"k8s.io/klog/v2"
 
-	drapbv1 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+	drapbv1alpha2 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
+	drapb "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
 )
 
 const PluginClientTimeout = 10 * time.Second
 
-type Client interface {
-	NodePrepareResource(
-		ctx context.Context,
-		namespace string,
-		claimUID types.UID,
-		claimName string,
-		resourceHandle string,
-	) (*drapbv1.NodePrepareResourceResponse, error)
-
-	NodeUnprepareResource(
-		ctx context.Context,
-		namespace string,
-		claimUID types.UID,
-		claimName string,
-		resourceHandle string,
-	) (*drapbv1.NodeUnprepareResourceResponse, error)
-}
-
 // Strongly typed address.
 type draAddr string
 
 // draPluginClient encapsulates all dra plugin methods.
 type draPluginClient struct {
-	pluginName          string
-	addr                draAddr
-	nodeV1ClientCreator nodeV1ClientCreator
+	pluginName        string
+	addr              draAddr
+	nodeClientCreator nodeClientCreator
 }
 
-var _ Client = &draPluginClient{}
+var _ drapb.NodeClient = &draPluginClient{}
 
-type nodeV1ClientCreator func(addr draAddr) (
-	nodeClient drapbv1.NodeClient,
+type nodeClientCreator func(addr draAddr) (
+	nodeClient drapb.NodeClient,
+	nodeClientOld drapbv1alpha2.NodeClient,
```

Comment:
I was wondering how we were going to handle the old API. This seems like a reasonable solution -- i.e. always make calls using the new API in `manager.go` but convert them under the hood to the old API (if necessary) from within the client.